tests/uc_oss/test_uc_oss_integration.py

```diff
@@ -0,0 +1,161 @@
+import os
+import subprocess
+import sys
+
+import pandas as pd
+import pytest
+from sklearn import datasets
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+
+import mlflow
+
+from tests.tracking.integration_test_utils import _await_server_up_or_die
+
+
+@pytest.mark.skipif(
+    "UC_OSS_INTEGRATION" not in os.environ,
+    reason="This test is only valid w/in the github workflow integration job",
+)
+@pytest.fixture(scope="module")
+def setup_servers():
+    with subprocess.Popen([sys.executable, "-m", "mlflow", "server", "--port", "5000"]) as proc:
+        try:
+            _await_server_up_or_die(5000)
+
+            mlflow_tracking_url = "http://127.0.0.1:5000"
+            uc_oss_url = "uc:http://127.0.0.1:8080"
+
+            mlflow.set_tracking_uri(mlflow_tracking_url)
+            mlflow.set_registry_uri(uc_oss_url)
+
+            yield mlflow_tracking_url
+        finally:
+            proc.terminate()
+
+
+def test_integration(setup_servers):
```

Comment:
Is it possible to split this into smaller sub tests?

Reply:
I'm not really sure what breaking this up would buy you.  I would still need to create a model and model version, and then delete the model and model version after every test.  I think there of integration tests as having a bit more latitude in what the test is accomplishing.